packages/docusaurus-plugin-content-docs/src/props.ts

```diff
@@ -52,7 +52,8 @@ Available document ids are:
       label: sidebarLabel || item.label || title,
       href: permalink,
       className: item.className,
-      customProps: item.customProps,
+      customProps:
+        item.customProps ?? docMetadata.frontMatter.sidebar_custom_props,
```

Comment:
Ooops... Just sat down to actually read the code. I don't think this is good. You should read the front matter during autogeneration (in `generator.ts`), since we are already converting a fully generated sidebar structure to what the client-side needs at this stage